app/scripts/controllers/paymentModal.js

```diff
@@ -350,11 +351,43 @@ angular
           });
       };
 
+      $scope.allPromotions = [
+        ...$scope.registration.globalPromotions,
+        ...$scope.registration.promotions,
+      ];
+
+      $scope.showAvailablePromotions = function () {
+        return (
```

Comment:
Thanks for refactoring this logic into the service! Now I'm wondering if we should just do `return availablePromotions.length > 0`?